docs/layouts/shortcodes/generated/pulsar_consumer_configuration.html

```diff
@@ -0,0 +1,180 @@
+        </tr>
+        <tr>
+            <td><h5>pulsar.consumer.deadLetterPolicy.deadLetterTopic</h5></td>
+            <td style="word-wrap: break-word;">(none)</td>
+            <td>String</td>
+            <td>Name of the dead topic where the failing messages will be sent.</td>
+        </tr>
+        <tr>
+            <td><h5>pulsar.consumer.deadLetterPolicy.maxRedeliverCount</h5></td>
+            <td style="word-wrap: break-word;">0</td>
+            <td>Integer</td>
+            <td>Maximum number of times that a message will be redelivered before being sent to the dead letter queue.</td>
+        </tr>
+        <tr>
+            <td><h5>pulsar.consumer.deadLetterPolicy.retryLetterTopic</h5></td>
+            <td style="word-wrap: break-word;">(none)</td>
+            <td>String</td>
+            <td>Name of the retry topic where the failing messages will be sent.</td>
+        </tr>
+        <tr>
+            <td><h5>pulsar.consumer.expireTimeOfIncompleteChunkedMessageMillis</h5></td>
+            <td style="word-wrap: break-word;">60000</td>
+            <td>Long</td>
+            <td>If producer fails to publish all the chunks of a message then consumer can expire incomplete chunks if consumer won't be able to receive all chunks in expire times (default 1 hour, in ms).</td>
+        </tr>
+        <tr>
+            <td><h5>pulsar.consumer.maxPendingChunkedMessage</h5></td>
+            <td style="word-wrap: break-word;">10</td>
+            <td>Integer</td>
+            <td>Consumer buffers chunk messages into memory until it receives all the chunks of the original message. While consuming chunk-messages, chunks from same message might not be contiguous in the stream and they might be mixed with other messages' chunks. So, consumer has to maintain multiple buffers to manage chunks coming from different messages. This mainly happens when multiple publishers are publishing messages on the topic concurrently or publisher failed to publish all chunks of the messages.<br />eg: M1-C1, M2-C1, M1-C2, M2-C2Messages M1-C1 and M1-C2 belong to original message M1, M2-C1 and M2-C2 messages belong to M2 message.<br />Buffering large number of outstanding uncompleted chunked messages can create memory pressure and it can be guarded by providing this <code class="highlighter-rouge">pulsar.consumer.maxPendingChunkedMessage</code> threshold. Once, consumer reaches this threshold, it drops the outstanding unchunked-messages by silently acking or asking broker to redeliver later by marking it unacked. This behavior can be controlled by configuration <code class="highlighter-rouge">pulsar.consumer.autoAckOldestChunkedMessageOnQueueFull</code></td>
```

Comment:
```suggestion
            <td>The consumer buffers chunk messages into memory until it receives all the chunks of the original message. While consuming chunk messages, chunks from the same message might not be contiguous in the stream and they might be mixed with other messages' chunks. So, the consumer has to maintain multiple buffers to manage chunks coming from different messages. This mainly happens when multiple publishers are publishing messages on the topic concurrently or publishers failed to publish all chunks of the messages.<br />For example, there are M1-C1, M2-C1, M1-C2, M2-C2 messages. M1-C1 and M1-C2 belong to the M1 original message while M2-C1 and M2-C2 belong to the M2 message.<br />Bufferingna large number of outstanding uncompleted chunked messages can bring memory pressure and it can be guarded by providing this <code class="highlighter-rouge">pulsar.consumer.maxPendingChunkedMessage</code> threshold. Once, a consumer reaches this threshold, it drops the outstanding unchunked messages by silently acknowledging or asking the broker to redeliver messages later by marking it unacknowledged. This behavior can be controlled by the <code class="highlighter-rouge">pulsar.consumer.autoAckOldestChunkedMessageOnQueueFull</code> option.</td>
```